core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -486,10 +488,10 @@ protected final GetResult getFromSearcher(Get get, Function<String, Searcher> se
     }
 
     public final GetResult get(Get get) throws EngineException {
```

Comment:
I'd pass `() -> fail("didn't expect a refresh")` when you don't expect a GET and do something a bit more elaborate when you expect a refresh:
```
AtomicBoolean refreshed = new AtomicBoolean(false);
engine.get(get, () -> refreshed.set(true));
assertTrue("expected a refresh", refreshed.get());
```